packages/vest/src/typings/vest.d.ts

```diff
@@ -167,6 +167,27 @@ interface ICreateResult {
   get: () => DraftResult;
   reset: () => void;
   remove: (fieldName: string) => void;
+  
+  hasErrors: (fieldName?: string) => boolean;
+  hasWarnings: (fieldName?: string) => boolean;
+  
+  getErrors(): { [fieldName: string]: string[] };
+  getErrors(fieldName: string): string[];
+
+  getWarnings(): { [fieldName: string]: string[] };
+  getWarnings(fieldName: string): string[];
+  
+  hasErrorsByGroup(groupName: string): boolean;
+  hasErrorsByGroup(groupName: string, fieldName: string): boolean;
+  
+  hasWarningsByGroup(groupName: string): boolean;
+  hasWarningsByGroup(groupName: string, fieldName: string): boolean;
+  
+  getErrorsByGroup(groupName: string): { [fieldName: string]: string[] };
+  getErrorsByGroup(groupName: string, fieldName: string): string[];
+  
+  getWarningsByGroup(groupName: string): { [fieldName: string]: string[] };
+  getWarningsByGroup(groupName: string, fieldName: string): string[];
```

Comment:
Nice!
Here's an enhancement idea: Since all those are already typed here https://github.com/ealush/vest/blob/latest/packages/vest/src/typings/vestResult.d.ts#L28-L85 why not move them to their own type and use in both places? Would make maintenance way easier

Reply:
Done.